topic-operator/src/main/java/io/strimzi/operator/topic/v2/ReplicasChangeClient.java

```diff
@@ -0,0 +1,373 @@
+    /**
+     * Create a new replicas change client instance.
+     * 
+     * @param config Topic Operator configuration.
+     */
+    public ReplicasChangeClient(TopicOperatorConfig config) {
+        this.config = config;
+        this.httpClientExecutor = Executors.newCachedThreadPool();
+        this.mapper = new ObjectMapper();
+    }
+
+    /**
+     * Send a topic_configuration request to create a task for replication factor change of one or more topics.
+     * This should be called when one or more .spec.replicas changes are detected.
+     * 
+     * @param reconcilableTopics Pending replicas changes.
+     * @return Replicas changes with status update.
+     */
+    public List<ReconcilableTopic> requestPendingChanges(List<ReconcilableTopic> reconcilableTopics) {
+        List<ReconcilableTopic> result = new ArrayList<>();
+        if (reconcilableTopics.isEmpty()) {
+            return result;
+        }
+        updateToPending(reconcilableTopics, "Replicas change pending");
+        result.addAll(reconcilableTopics);
+        
+        try {
+            LOGGER.debugOp("Sending topic configuration request, topics {}", topicNames(reconcilableTopics));
+            HttpClient client = buildHttpClient();
+            client.sendAsync(buildTopicConfigPostRequest(reconcilableTopics), HttpResponse.BodyHandlers.ofString())
+                .thenAccept(response -> {
+                    if (response.statusCode() != 200) {
+                        updateToFailed(result, format("Replicas change failed (%s)", response.statusCode()), response);
+                        return;
+                    }
+                    try {
+                        String userTaskId = response.headers().firstValue(USER_TASK_ID_HEADER).get();
+                        updateToOngoing(result, "Replicas change ongoing", userTaskId);
+                    } catch (Throwable e) {
+                        updateToFailed(result, format("Failed to get %s header", USER_TASK_ID_HEADER), response);
+                    }
+                }).join();
+        } catch (Throwable t) {
+            updateToFailed(result, format("Replicas change failed, %s", t.getMessage()), null);
+        }
+        return result;
+    }
+
+    /**
+     * Send a user_tasks request to check the state of ongoing replication factor changes.
+     * This should be called periodically to update the active tasks cache and KafkaTopic status.
+     *
+     * @param reconcilableTopics Ongoing replicas changes.
+     * @return Replicas changes with status update.
+     */
+    public List<ReconcilableTopic> requestOngoingChanges(List<ReconcilableTopic> reconcilableTopics) {
+        List<ReconcilableTopic> result = new ArrayList<>();
+        if (reconcilableTopics.isEmpty()) {
+            return result;
+        }
+        result.addAll(reconcilableTopics);
+        
+        Map<String, List<ReconcilableTopic>> groupByUserTaskId = new HashMap<>();
+        reconcilableTopics.forEach(rt -> {
```

Comment:
Personally I really hate `forEach`. I don't really know what the point of it is as an API. I think either you should code this as a normal `for` statement, or use the Streams API and code this as a `filter(...).map()....collect(Collectors.groupingBy(...))`. Personally I'd do the latter, because I think it's easier to get a graph of what's happening that way.